test/interpreter/function-slope.spec.ts

```diff
@@ -0,0 +1,93 @@
+import {HyperFormula} from '../../src'
+import {ErrorType} from '../../src/Cell'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('SLOPE', () => {
+  it('validates number of arguments',  () => {
+    const engine =  HyperFormula.buildFromArray([
+      ['=SLOPE(B1:B5)'],
+      ['=SLOPE(B1:B5, C1:C5, D1:D5)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('ranges need to have same amount of elements',  () => {
+    const engine =  HyperFormula.buildFromArray([
+      ['=SLOPE(B1:B5, C1:C6)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.EqualLength))
```

Comment:
When they are of the same length but empty, there should be different error message
"If known_y's and known_x's **are empty** or have a different number of data points, SLOPE returns the #N/A error value."

Reply:
as above